### ISSUE TITLE
Internal error in clause compiler

### ISSUE THREAD

**Issue by nad** [agda/agda#4259]
The following code triggers an internal error for Agda 2.6.1-cf76109, but is accepted by Agda 2.6.0.1:
```agda
module _ where

data D : Set where
  d : D

postulate
  I : Set

data P (i : I) : Set where
  p : P i → P i

data Q (i : I) : P i → Set where
  q : (x : P i) → Q i (p x)

f : (i : I) → P i → Set₁
f i x = Set
  where
  g : Q _ x → Set₁
  g (q _) = Set
```
```
An internal error has occurred. Please report this as a bug.
Location of the error: src/full/Agda/TypeChecking/CompiledClause/Compile.hs:121
```
Possibly related: #4254.

**Comment by UlfNorell** [comment]
Goes away with `--no-forcing` so looks like a problem with the new forcing translation.

**Comment by UlfNorell** [comment]
Looks like the problem is that I didn't understand this code:

https://github.com/agda/agda/blob/ea4c2247b69f60846aca78a1d1ae675609d7ee50/src/full/Agda/TypeChecking/Rules/LHS/Unify.hs#L994-L997

ping @jespercockx

**Comment by UlfNorell** [comment]
Ok, so fix is here #4278.

My understanding of the code is that if a solution step fails ("Cannot solve variable...") we normalise the solution and the variable telescope and try again. The problem was that the trying again only did `solveVar` skipping the forcing translation step. The fix reruns the entire `Solution` step from the top.
